### PR TITLE
Yifan-project-library-solo

### PR DESCRIPTION
**Pull request by Yifan-858**
## Netlify link
https://yifan-pokectbookshop.netlify.app/

## Collaborators
[Yifan-858]

### CODE REVIEW

File: books-refractored.js [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+  } else {
+    console.log("something is wrong");
+  } // get the intersection of filteredBooks array and newFilterBooks
+
+  booksToDisplay.length === 0
+    ? console.log("no result")
+    : console.log("there are books left");
+
+  previousFilter = filterType;
+};
```
Comment (lunek1) [inline books-refractored.js:367]: Would this work better? If you combine `isGenreFiltered` and `isYearFiltered`.

const checkFiltered = () => {
  if (!isGenreFiltered && !isYearFiltered) {
    booksToDisplay = books;
  } else if (isGenreFiltered && !isYearFiltered) {
    booksToDisplay = filteredBooks;
  } else if (!isGenreFiltered && isYearFiltered) {
    booksToDisplay = filteredBooks;
  } else if (isGenreFiltered && isYearFiltered) {
    booksToDisplay = genreFilteredBooks.filter((book) =>
      yearFilteredBooks.includes(book)
    );
  }

  previousFilter = filterType;
};

File: books-refractored.js [hunk 2]
```diff
@@ -0,0 +1,675 @@
...
+        <p class="book-genre"><span class="tag">Genre :</span> ${book.genre}</p>
+        <p class="book-rating"><span class="tag">Rating :</span> ${book.rating}</p>
+        <p class="book-description">${book.description}</p>
+      </div>
+    </div>
+    `;
+  });
+};
+
+bookCards(books); //Have all the books displayed when the page is loaded
```
Comment (lunek1) [inline books-refractored.js:298]: Very clear and efficent function for the bookcards. 🙌

File: books-refractored.js [hunk 3]
```diff
@@ -0,0 +1,675 @@
...
+
+    if (stringA < stringB) {
+      return 1;
+    } else if (stringA > stringB) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
```
Comment (lunek1) [inline books-refractored.js:460]: I see you have `console.log` statements in parts of the code. They are not needed when project is done. ✨

File: books-refractored.js [hunk 4]
```diff
@@ -0,0 +1,675 @@
...
+
+  genreBtn.addEventListener("mouseout", () => {
+    genreBtn.classList.remove("hover");
+  });
+
+  genreBtn.addEventListener("click", () => {
+    genreBtn.classList.remove("hover");
+    toggleGenreSelected(genreBtn);
+  });
+});
```
Comment (lunek1) [inline books-refractored.js:525]: The mouseover and mouseout event listeners on the book cards adds a nice touch. Makes the user experience smoother! Nice! ✨

File: books-refractored.js [hunk 5]
```diff
@@ -0,0 +1,675 @@
...
+
+reset.addEventListener("click", () => {
+  reset.classList.add("selected");
+  setTimeout(() => {
+    reset.classList.remove("selected");
+  }, 800);
+  bookCards(books);
+  resetFunc();
+});
```
Comment (lunek1) [inline books-refractored.js:675]: Very clever to add a reset button to clear the user input and also adding a short delay for visual clarity. 👏

**Review by lunek1 — COMMENTED** [review]
Your site looks really good and it works flawless. From what I can tell your site met the requirements. 

I really like the simplicity in your code for the logic in functions and it works well.

I would have wanted more comments to easier understand the logic behind some of the functions. Some of it had good comments though that were easy to understand.

Regarding responsiveness:
- Containers are responding well to different screens but they are not centered. Centering them would make it more visually appealing for the user. As of now there are gaps on the sides of the containers that varies. The gap/padding to the left is consistent but when you change between screens you can see the gap to the right isn't consistent with the left. 
- In mobile view maybe a dropdown would be suitable for filter and sorting? Making it look more neat.

Again, great job on the project! 🥳

File: index.html
Comment (NellaMori) [inline index.html:1]: Please format your code to have the right spaces and indentation 👍

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,2 @@
+books.js
+recipes.js
```
Comment (NellaMori) [inline .gitignore:2]: why are these 2 files in your .gitignore? If they are not needed go ahead and delete them 🧹

**Review by NellaMori — APPROVED** [review]
Hi Yifan, well done finishing your Library app 📚 You followed the instructions and the app fulfils the requirements of filtering and responsiveness. Some improvements can be done regarding the UI and UX: the buttons take all the space in the screen when ideally you want, by clicking on them, to see the result of the filtering and sorting. Right now you need to scroll to see the books, which is not good. Also, in mobile mode, the book card should take up as much space as possible in the screen, now there's a lot of empty space. Remove the empty unused spaces in the app to improve the UI and you're good to go.I left some comments in the code to improve the UI, but overall you did a good job 🚀

**Pull request closed without merging**